File: VB/06_Annotations/ModifyLineAnnotation.vb
Imports Spire.Pdf
Imports Spire.Pdf.Annotations

Namespace ModifyLineAnnotation
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Create a new PdfDocument object 
            Dim document As New PdfDocument()

            ' Load an existing PDF document
            document.LoadFromFile("..\..\..\..\..\..\Data\PdfLineAnnotation.pdf")

            ' Get the first annotation from the first page of the document
            Dim annotation As PdfAnnotation = document.Pages(0).Annotations(0)

            ' Check if the annotation is a PdfLineAnnotationWidget
            If TypeOf annotation Is PdfLineAnnotationWidget Then
                ' Cast the annotation as a PdfLineAnnotationWidget
                Dim lineAnn As PdfLineAnnotationWidget = TryCast(annotation, PdfLineAnnotationWidget)

                ' Modify the author and subject properties of the line annotation
                lineAnn.Author = "Author_test"
                lineAnn.Subject = "Subject_test"
            End If

            ' Save the modified document to a file
            Dim result As String = "ModifyLineAnnotation.pdf"
            document.SaveToFile(result)

            ' Close the document
            document.Close()

            ' Launch the Pdf file
            PDFDocumentViewer(result)
        End Sub

        Private Sub PDFDocumentViewer(ByVal filename As String)
            Try
                Process.Start(filename)
            Catch
            End Try
        End Sub
    End Class
End Namespace

File: CS/06_Annotations/ModifyLineAnnotation.cs
using Spire.Pdf;
using Spire.Pdf.Annotations;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace ModifyLineAnnotation
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            string result = "ModifyLineAnnotation.pdf";

            // Create a new PdfDocument object
            PdfDocument document = new PdfDocument();

            // Load an existing PDF file into the document from the specified path
            document.LoadFromFile(@"..\..\..\..\..\..\Data\PdfLineAnnotation.pdf");

            // Retrieve the first annotation widget from the first page of the document
            PdfAnnotation annotation = document.Pages[0].Annotations[0];

            // Check if the retrieved annotation is a PdfLineAnnotationWidget
            if (annotation is PdfLineAnnotationWidget)
            {
                // Cast the annotation as a PdfLineAnnotationWidget to access its specific properties and methods
                PdfLineAnnotationWidget lineAnn = annotation as PdfLineAnnotationWidget;

                // Set the author property of the line annotation to "Author_test"
                lineAnn.Author = "Author_test";

                // Set the subject property of the line annotation to "Subject_test"
                lineAnn.Subject = "Subject_test";
            }

            // Save the modified document to the specified file path
            document.SaveToFile(result);

            //Launch the Pdf file
            PDFDocumentViewer(result);
        }

        private void PDFDocumentViewer(string filename)
        {
            try
            {
                System.Diagnostics.Process.Start(filename);
            }
            catch { }
        }
    }
}
